hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterClientProtocol.java

```diff
@@ -641,6 +646,11 @@ public void rename2(final String src, final String dst,
         new Class<?>[] {String.class, String.class, options.getClass()},
         new RemoteParam(), dstParam, options);
     if (isMultiDestDirectory(src)) {
+      if (locs.size() == srcLocations.size()) {
+        throw new IOException(
+            "Rename of " + src + " to " + dst + " is not allowed," +
+                " The remote location should be exactly same.");
```

Comment:
What's the assumption on why comparing sizes is equal to saying it is the same?

Reply:
This exception message is very confusing. Unless I went back and checked the description, wasn't able to decode the intent.
Can we change it to 
```
throw new IOException(
            "Rename of " + src + " to " + dst + " is not allowed," +
                " The number of remote locations for both source and target should be same.");
```